download-utilities.sh

```diff
@@ -718,9 +740,27 @@ process_args() {
         desired_command="${1#--}"
         shift
         ;;
+      --debug-yaml)
+        debug_yaml=true
+        # Check if next argument exists and doesn't start with --
+        if [ $# -gt 1 ] && [[ ! "$2" =~ ^-- ]]; then
+          debug_yaml_file="$2"
+          shift
```

Comment:
```suggestion
          shift
        else
          echo "ERROR: A file name is required after --debug-yaml" >&2
          exit 1
```

This helps prevent continuing if the user intended an option but provided an invalid one.